addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/discovery/SatelDeviceDiscoveryService.java

```diff
@@ -0,0 +1,181 @@
+ * devices and modules connected to the API console
+ *
+ * @author Krzysztof Goworek - Initial contribution
+ *
+ */
+public class SatelDeviceDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int OUTPUT_FUNCTION_SHUTTER = 105;
+
+    private final Logger logger = LoggerFactory.getLogger(SatelDeviceDiscoveryService.class);
+
+    private SatelBridgeHandler bridgeHandler;
+    private BundleContext bundleContext;
+    private volatile boolean scanStopped;
+
+    public SatelDeviceDiscoveryService(SatelBridgeHandler bridgeHandler, BundleContext bundleContext)
+            throws IllegalArgumentException {
+        super(Sets.union(DEVICE_THING_TYPES_UIDS, VIRTUAL_THING_TYPES_UIDS), 60, true);
+        this.bridgeHandler = bridgeHandler;
+        this.bundleContext = bundleContext;
+    }
+
+    @Override
+    protected void startScan() {
+        try (BundleTextTranslator translator = new BundleTextTranslator(bundleContext)) {
+            scanStopped = false;
+            if (bridgeHandler.isInitialized()) {
+                addThing(THING_TYPE_SYSTEM, null, translator.getText("system.label", "System"), Collections.emptyMap());
+            }
+            if (!scanStopped) {
+                scanForDevices(DeviceType.PARTITION_WITH_OBJECT, bridgeHandler.getIntegraType().getPartitions());
+            }
+            if (!scanStopped) {
+                scanForDevices(DeviceType.ZONE_WITH_PARTITION, bridgeHandler.getIntegraType().getZones());
+            }
+            if (!scanStopped) {
+                scanForDevices(DeviceType.OUTPUT, bridgeHandler.getIntegraType().getZones());
+            }
+        }
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        scanStopped = true;
+        super.stopScan();
+    }
+
+    private void scanForDevices(DeviceType deviceType, int maxNumber) {
+        logger.debug("Scanning for {} started", deviceType.name());
+        for (int i = 1; i <= maxNumber && !scanStopped; ++i) {
+            try {
+                ReadDeviceInfoCommand cmd = new ReadDeviceInfoCommand(deviceType, i);
+                cmd.ignoreResponseError();
+                if (bridgeHandler.sendCommand(cmd, false)) {
+                    String name = cmd.getName(bridgeHandler.getEncoding());
+                    int deviceKind = cmd.getDeviceKind();
+                    logger.debug("Found device: type={}, id={}, name={}, kind/function={}", deviceType.name(), i, name,
+                            deviceKind);
+                    if (isDeviceAvailable(deviceType, deviceKind)) {
+                        addDevice(deviceType, deviceKind, i, name);
+                    }
+                } else if (cmd.getState() != SatelCommand.State.FAILED) {
+                    // serious failure, disconnection or so
+                    scanStopped = true;
+                    logger.error("Unexpected failure during scan for {} using {}", deviceType.name(),
```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. 

Please change this to use the correct logging levels and if possible change the ONLINE/OFFLINE status of thing / bridge.

Reply:
Please check others as well